bigtable/client/data.cc

```diff
@@ -75,5 +73,32 @@ void Table::Apply(SingleRowMutation&& mut) {
   }
 }
 
+void Table::BulkApply(BulkMutation&& mut) {
+  auto backoff_policy = rpc_backoff_policy_->clone();
+  auto retry_policy = rpc_retry_policy_->clone();
+  auto idemponent_policy = idempotent_mutation_policy_->clone();
+
+  detail::BulkMutator mutator(table_name_, *idemponent_policy,
+                              std::forward<BulkMutation>(mut));
+
+  while (mutator.HasPendingMutations()) {
+    grpc::ClientContext client_context;
+    backoff_policy->setup(client_context);
+    retry_policy->setup(client_context);
+
+    auto status = mutator.MakeOneRequest(client_->Stub(), client_context);
+    if (not status.ok() and not retry_policy->on_failure(status)) {
+      break;
+    }
+    auto delay = backoff_policy->on_completion(status);
+    std::this_thread::sleep_for(delay);
+  }
+  auto failures = mutator.ExtractFinalFailures();
+  if (not failures.empty()) {
+    throw PermanentMutationFailure("Permanent errors in Table::BulkApply()",
```

Comment:
The overall status needs to be included here as well.  Situations like table/cluster does not exist, or there are 100K mutations need to be surfaced.

Reply:
Looks like this was done in [this commit](https://github.com/GoogleCloudPlatform/google-cloud-cpp/pull/73/commits/efbdf581908c69b40597cae7fe4cf28984e21991).

Reply:
Correct, sorry I forgot to update this review comment.